src/utils.ts

```diff
@@ -4,7 +4,37 @@ import type {
   CommonBlockProps,
   FileBlockProps,
   FolderBlockProps,
-} from "@utils";
+} from "../utils/types";
+
+import { Endpoints, OctokitResponse, RequestParameters } from "@octokit/types";
+import { endpoint } from "@octokit/endpoint";
+
+type FilterConditionally<Source, Condition> = Pick<
+  Source,
+  { [K in keyof Source]: K extends Condition ? K : never }[keyof Source]
+>;
+type GetEndpoints = FilterConditionally<Endpoints, `GET ${string}`>;
+
+export async function onRequestGitHubData<
+  Endpoint extends keyof GetEndpoints,
+  EndpointParameters extends GetEndpoints[Endpoint]["parameters"]
+>(
+  route: Endpoint,
+  parameters?: EndpointParameters & RequestParameters
+): Promise<OctokitResponse<Endpoints[Endpoint]>["data"]["response"]["data"]> {
+  const params = endpoint<Endpoint, EndpointParameters>(route, parameters);
+
+  if (params.method !== "GET") {
+    throw new Error("Only GET requests are supported.");
+  }
+
+  return makeRequest("onRequestGitHubData", {
+    path: params.url,
+    ...params,
```

Comment:
only `path` is used on the `block-platform` side, I guess `endpoint` returns a `params.url` that contains the API call parameters already?

Reply:
Good call. `params` here includes other details like `headers` and `mediaType`, but I'm not sure if we want to support downstream customization of these options.

Reply:
Also, if we want to prevent against non-GET requests from `blocks-platform`, we might want to send along the whole `params` object since it has a `method` key on it.